packages/eslint-plugin/src/rules/space-infix-ops.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  AST_NODE_TYPES,
+  TSESLint,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import baseRule from 'eslint/lib/rules/space-infix-ops';
+import { TSEnumMember } from '../../../parser/node_modules/@typescript-eslint/types/dist/ts-estree';
+import * as util from '../util';
+
+export type Options = util.InferOptionsTypeFromRule<typeof baseRule>;
+export type MessageIds = util.InferMessageIdsTypeFromRule<typeof baseRule>;
+
+export default util.createRule<Options, MessageIds>({
+  name: 'space-infix-ops',
+  meta: {
+    type: 'layout',
+    docs: {
+      description:
+        'This rule is aimed at ensuring there are spaces around infix operators.',
+      category: 'Stylistic Issues',
+      recommended: false,
+      extendsBaseRule: true,
+    },
+    fixable: baseRule.meta.fixable,
+    schema: baseRule.meta.schema,
+    messages: baseRule.meta.messages,
+  },
+  defaultOptions: [
+    {
+      int32Hint: false,
+    },
+  ],
+  create(context) {
+    const rules = baseRule.create(context);
+    const sourceCode = context.getSourceCode();
+
+    /**
+     * Find the first non space specified char
+     * @param {TSESTree.token} left The token to the left
+     * @param {TSESTree.Token} right The token to the right
+     * @param {string} op The token to find
+     * @returns {TSESTree.Token|null}
+     * @private
+     */
+    function getFirstNonSpacedToken(
+      left: TSESTree.Token,
+      right: TSESTree.Token,
+      op: string,
+    ): TSESTree.Token | null {
+      const operator = sourceCode.getFirstTokenBetween(
+        left,
+        right,
+        token => token.value === op,
+      );
+      const prev = sourceCode.getTokenBefore(operator!);
+      const next = sourceCode.getTokenAfter(operator!);
+
+      if (
+        !sourceCode.isSpaceBetweenTokens(prev!, operator!) ||
+        !sourceCode.isSpaceBetweenTokens(operator!, next!)
+      ) {
+        return operator;
+      }
+
+      return null;
+    }
```

Comment:
you could inline this along with the report - right now you're recomputing `sourceCode.getTokenBefore` and `sourceCode.getTokenAfter` when you could just compute it once